Framework/API/test/MatrixWorkspaceTest.h

```diff
@@ -1732,7 +1732,114 @@ class MatrixWorkspaceTest : public CxxTest::TestSuite {
     checkDetectorSignedTwoTheta(Geometry::X, {{1., -1., -1., 1.}});
   }
 
+  void
+  test_that_binIndexOf_returns_the_correct_index_for_a_histogram_workspace_when_the_x_bins_are_in_ascending_order() {
+    std::vector<double> const xValues{1.0, 2.0, 3.0, 4.0};
+    auto const workspace = getWorkspaceWithPopulatedX(1, 4, 3, xValues);
+
+    TS_ASSERT_EQUALS(workspace.binIndexOf(3.0), 1);
```

Comment:
It's good we have more tests around this now. Could we add some different values for each of them so that we're not just testing the boundaries? I think the tests in `testBinIndexOf` could just be moved here?